src/Avalonia.Base/Input/GestureRecognizers/ScrollGestureRecognizer.cs

```diff
@@ -181,53 +191,75 @@ protected override void PointerReleased(PointerReleasedEventArgs e)
-                        lastTime = st.Elapsed;
-
-                        var speed = _inertia * Math.Pow(InertialResistance, st.Elapsed.TotalSeconds);
-                        var distance = speed * elapsedSinceLastTick.TotalSeconds;
-                        var scrollGestureEventArgs = new ScrollGestureEventArgs(_gestureId, distance);
-                        Target!.RaiseEvent(scrollGestureEventArgs);
-
-                        if (!scrollGestureEventArgs.Handled || scrollGestureEventArgs.ShouldEndScrollGesture)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-
-                        // EndGesture using InertialScrollSpeedEnd only in the direction of scrolling
-                        if (CanVerticallyScroll && CanHorizontallyScroll && Math.Abs(speed.X) < InertialScrollSpeedEnd && Math.Abs(speed.Y) <= InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-                        else if (CanVerticallyScroll && Math.Abs(speed.Y) <= InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-                        else if (CanHorizontallyScroll && Math.Abs(speed.X) < InertialScrollSpeedEnd)
-                        {
-                            EndGesture();
-                            return false;
-                        }
-
-                        return true;
-                    }, TimeSpan.FromMilliseconds(16), DispatcherPriority.Background);
+
+                    MediaContext.Instance.RequestAnimationFrame(StartInertia);
                 }
             }
         }
+
+        private void StartInertia(TimeSpan timeSpan)
+        {
+            if (_inertia is { } inertia)
+            {
+                _lastTime = timeSpan;
+                _inertiaStartTime = timeSpan;
+                _currentInertiaGestureId = _gestureId;
+                Target!.RaiseEvent(new ScrollGestureInertiaStartingEventArgs(_gestureId, inertia));
+
+                MediaContext.Instance.RequestAnimationFrame(OnAnimationRequested);
+            }
+        }
+
+        private void OnAnimationRequested(TimeSpan timeSpan)
+        {
+            // Another gesture has started, finish the current one
+            if (_gestureId != _currentInertiaGestureId || _inertia == null)
+            {
+                _inertia = null;
+                return;
+            }
+
+            var elapsedSinceLastTick = timeSpan - _lastTime;
+            _lastTime = timeSpan;
+
+            var speed = _inertia * Math.Pow(InertialResistance, (_lastTime - _inertiaStartTime).TotalSeconds);
+            var distance = speed * elapsedSinceLastTick.TotalSeconds;
+            _inertiaDispatchOperation = Dispatcher.UIThread.InvokeAsync(() =>
```

Comment:
Can you add a comment here, that we actually _want_ InvokeAsync only to raise events with Input priority?

Reply:
IIRC from the discussion, priority change was the only reason.

Reply:
Added.